java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/websocket/GenericMqttWebSocketPublisher.java

```diff
@@ -0,0 +1,84 @@
+package com.cumulocity.generic.mqtt.client.websocket;
+
+import com.cumulocity.generic.mqtt.client.GenericMqttConnectionConfig;
+import com.cumulocity.generic.mqtt.client.GenericMqttPublisher;
+import com.cumulocity.generic.mqtt.client.converter.GenericMqttMessageConverter;
+import com.cumulocity.generic.mqtt.client.model.GenericMqttMessage;
+import com.cumulocity.generic.mqtt.client.websocket.exception.GenericMqttWebSocketClientException;
+import com.cumulocity.rest.representation.reliable.notification.NotificationTokenRequestRepresentation;
+import com.cumulocity.sdk.client.messaging.notifications.TokenApi;
+import org.apache.commons.codec.binary.Base64;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+
+class GenericMqttWebSocketPublisher implements GenericMqttPublisher {
+
+    private final static String WEBSOCKET_URL_PATTERN = "%s/notification2/producer/?token=%s";
+    private static final String TOKEN_TYPE = "mqtt";
+
+    private final String webSocketBaseUrl;
+    private final TokenApi tokenApi;
+
+    private final GenericMqttMessageConverter genericMqttMessageConverter = new GenericMqttMessageConverter();
+
+    private final AtomicInteger sequence = new AtomicInteger();
+    private final GenericMqttConnectionConfig config;
+
+    private GenericMqttWebSocketClient producer;
+
+    public GenericMqttWebSocketPublisher(String webSocketBaseUrl, TokenApi tokenApi, GenericMqttConnectionConfig config) {
+        this.webSocketBaseUrl = webSocketBaseUrl;
+        this.tokenApi = tokenApi;
+        this.config = config;
+    }
+
+    @Override
+    public void publish(GenericMqttMessage genericMqttMessage) {
+        final NotificationTokenRequestRepresentation tokenRequestRepresentation = getTokenRepresentation(config.getTopic());
```

Comment:
issue: do we create a Token for each message published?

Also see comment below about duration being documented. Unless I don't understand the scope, I'd guess the Token should be created at instantiation time (and ideally the duration should be able to be set but user)

Reply:
Make sense but it is not really critical imo as the push messages won't be used very often, probably only for firmware upgrade. Maybe use `Suppliers.memoize`?

Reply:
Done.